tests/test_pipeline.py

```diff
@@ -48,32 +48,52 @@ def name(self) -> str:
         return "test-source"
 
     def __del__(self):
-        self._destructor_cb()
+        if self._destructor_cb is not None:
+            self._destructor_cb()
 
 
 class SinkTestStage(InMemorySinkStage):
 
-    def __init__(self, config, destructor_cb: typing.Callable[[], None]):
+    def __init__(self,
+                 config,
+                 on_start_cb: typing.Callable[[], None] = None,
+                 start_async_cb: typing.Callable[[], None] = None,
+                 destructor_cb: typing.Callable[[], None] = None):
         super().__init__(config)
+        self._on_start_cb = on_start_cb
+        self._start_async_cb = start_async_cb
         self._destructor_cb = destructor_cb
 
     @property
     def name(self) -> str:
         return "test-sink"
 
+    def on_start(self):
+        if self._on_start_cb is not None:
+            self._on_start_cb()
+
+    async def start_async(self):
+        await super().start_async()
+        if self._start_async_cb is not None:
+            self._start_async_cb()
```

Comment:
Question: should this be awaited?

Reply:
I double-checked, in this case `_start_async_cb` is a regular blocking method.